backend/main/src/Services/DeleteUserService/DeleteUserService.ts

```diff
@@ -0,0 +1,165 @@
+import {
+	BadRequestStatus,
+	Controller,
+	InternalServerErrorStatus,
+} from "src/Shared";
+import { IGroupRepository } from "src/Group/Domain/IGroupRepository";
+import { FindGroupsInformationQuery } from "src/Group/Query/FindGroupsInformation/FindGroupsInformatoinQuery";
+import { UserId } from "src/User";
+import { Group, GroupId } from "src/Group/Domain/Entities/Group";
+import { IUserRepository } from "src/User/Domain/IUserRepository";
+import {
+	AdminDeleteUserCommand,
+	AdminDeleteUserCommandInput,
+	CognitoIdentityProviderClient,
+	ListUsersCommand,
+	ListUsersCommandInput,
+} from "@aws-sdk/client-cognito-identity-provider";
+import { COGNITO_USER_POOL_ID } from "src/Settings/Setting";
+import { Err, Ok, Result } from "result-ts-type";
+
+interface IDeleteUserProps {
+	/** requested by the user */
+	userId: string;
+	/** from the token */
+	targetUserId: string;
+}
+
+/**
+ * Delete a user.
+ * The groups and containers that are possessed by only this user.
+ * This service also deletes the user from the Cognito User Pool.
+ */
+export class DeleteUserService extends Controller<IDeleteUserProps> {
+	constructor(
+		readonly groupRepository: IGroupRepository,
+		readonly userRepository: IUserRepository,
+		readonly groupQuery: FindGroupsInformationQuery,
+	) {
+		super();
+	}
+
+	async handler(input: IDeleteUserProps) {
+		// check the requested user token and the user token's one
```

Comment:
I feel it's better to use the word user ID cuz the comparison is not about tokens but IDs.
```suggestion
		// check the requested user ID is the same as the user ID from the auth token.
```